pycaret/internal/metrics.py

```diff
@@ -87,12 +136,15 @@ def __call__(self, y_true, y_pred, **kwargs):
         return self.score_func(y_true, y_pred, **kwargs)
 
 
-class _ThresholdScorerWithErrorScore(_ThresholdScorer):
-    def __init__(self, score_func, sign, kwargs, error_score=np.nan):
+class ScorerWithErrorScore(_Scorer):
+    def __init__(
+        self, score_func, sign, kwargs, error_score=np.nan, response_method="predict"
+    ):
         super().__init__(score_func=score_func, sign=sign, kwargs=kwargs)
         self.error_score = error_score
+        self._response_method = response_method
```

Comment:
I think that you need to provide the `response_method` to the scorer.

```python
    def __init__(
        self, score_func, sign, kwargs, error_score=np.nan, response_method="predict"
    ):
        super().__init__(
            score_func=score_func, sign=sign, kwargs=kwargs, response_method=response_method
        )
        self.error_score = error_score
```

Reply:
still not working, have errors!

Reply:
fixed, i think... now other warnings...
 UserWarning: Note that pos_label (set to 'MM') is ignored when average != 'binary' (got 'weighted'). You may use labels=[pos_label] to specify a single positive class.

 UserWarning: X does not have valid feature names, but RandomForestClassifier was fitted with feature names